doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -360,38 +360,75 @@ int main(int argc, char** argv)
   visual_tools.trigger();
   visual_tools.prompt("next step");
 
-  // Now, let's attach the collision object to the robot.
+  // Attaching Objects
+  // ^^^^^^^^^^^^^^^^^
+  //
+  // Now, let's attach a different type of collision object to the robot, for the purposes of "manipulating" it.
+  moveit_msgs::CollisionObject attachment_object;
+  attachment_object.id = "box2";
+
+  shape_msgs::SolidPrimitive another_primitive;
+  another_primitive.type = primitive.BOX;
+  another_primitive.dimensions.resize(3);
+  another_primitive.dimensions[0] = 0.1;
+  another_primitive.dimensions[1] = 0.1;
+  another_primitive.dimensions[2] = 0.1;
+
+  // We define the frame/pose for this box so that it appears in the gripper
+  attachment_object.header.frame_id = move_group.getEndEffectorLink();
+  geometry_msgs::Pose grab_pose;
+  grab_pose.orientation.w = 1.0;
+  grab_pose.position.z = 0.07;
+
+  // Now we go through the process to add it to the world
+  attachment_object.primitives.push_back(another_primitive);
+  attachment_object.primitive_poses.push_back(grab_pose);
+  attachment_object.operation = attachment_object.ADD;
+
+  std::vector<moveit_msgs::CollisionObject> attachment_objects;
+  attachment_objects.push_back(attachment_object);
+  planning_scene_interface.addCollisionObjects(attachment_objects);
```

Comment:
You can use [applyCollisionObject](http://docs.ros.org/api/moveit_ros_planning_interface/html/classmoveit_1_1planning__interface_1_1PlanningSceneInterface.html#a9f89682bc0bc4e85d3317c2f3558f075) to avoid making the vector. The user encounters the function at the end when both objects are removed from the scene.

I would also add a remark that you can use [applyAttachedCollisionObject](http://docs.ros.org/api/moveit_ros_planning_interface/html/classmoveit_1_1planning__interface_1_1PlanningSceneInterface.html#a08cbe97d3d33d13f568a7659283c3137) to attach an object to the robot directly.